pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -118,11 +121,70 @@ func (pl *VolumeBinding) EventsToRegister() []framework.ClusterEventWithHint {
 		// When CSIStorageCapacity is enabled, pods may become schedulable
 		// on CSI driver & storage capacity changes.
 		{Event: framework.ClusterEvent{Resource: framework.CSIDriver, ActionType: framework.Add | framework.Update}},
-		{Event: framework.ClusterEvent{Resource: framework.CSIStorageCapacity, ActionType: framework.Add | framework.Update}},
+		{Event: framework.ClusterEvent{Resource: framework.CSIStorageCapacity, ActionType: framework.Add | framework.Update}, QueueingHintFn: pl.isSchedulableAfterCSIStorageCapacityChange},
 	}
 	return events
 }
 
+func (pl *VolumeBinding) isSchedulableAfterCSIStorageCapacityChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	oldCap, newCap, err := util.As[*storagev1.CSIStorageCapacity](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	logger = klog.LoggerWithValues(
+		logger,
+		"Pod", klog.KObj(pod),
+		"CSIStorageCapacity", klog.KObj(newCap),
+	)
+
+	for _, vol := range pod.Spec.Volumes {
+		var pvc *v1.PersistentVolumeClaim
+		switch {
+		case vol.PersistentVolumeClaim != nil:
+			pvcName := vol.PersistentVolumeClaim.ClaimName
+			pvc, err = pl.PVCLister.PersistentVolumeClaims(pod.Namespace).Get(pvcName)
+			if err != nil {
+				return framework.Queue, err
+			}
+		case vol.Ephemeral != nil:
+			pvc = &v1.PersistentVolumeClaim{
+				ObjectMeta: vol.Ephemeral.VolumeClaimTemplate.ObjectMeta,
+				Spec:       vol.Ephemeral.VolumeClaimTemplate.Spec,
+			}
+		default:
+			continue
+		}
+
+		if pvc.Spec.VolumeName != "" {
+			// Skipping the check for CSIStorageCapacity as the PVC is configured
+			// to be bound to an existing PV.
+			continue
+		}
+
+		className := volume.GetPersistentVolumeClaimClass(pvc)
+		if newCap.StorageClassName == className {
```

Comment:
better to continue early instead of a large `if newCap.StorageClassName == className {...}`.

```suggestion
		if newCap.StorageClassName != className {
		  continue
		}
```